### PR TITLE
refactor: render React the right way and improve TypeScript

### LINKED ISSUE #60: Wrong way of rendering React Components
It seems this line renders React Components in the wrong way.
Is this intended or did I miss some context?! 🤔 

https://github.com/Hashnode/starter-kit/blob/c06bebf8e29ab4251d8892248b61a3c829fbe46e/packages/blog-starter-kit/themes/personal/pages/%5Bslug%5D.tsx#L177

Should be something like this

```tsx
{post ? <Post publication={publication} post={post /> : <Page page={page} />}
```

The problem is explained in this video: https://youtu.be/QuLfCUh-iwI?t=25

### PR DESCRIPTION
**Pull request by iamandrewluca**
- [x] Enterprise
- [x] Personal
- [x] Hashnode

---

- The first commit does formatting with prettier for files that will be changed
- The second commit does the changes.

---

Closes #60

### CODE REVIEW

File: packages/blog-starter-kit/themes/personal/pages/[slug].tsx [hunk 1]
```diff
@@ -191,63 +189,45 @@ export const getStaticProps: GetStaticProps<Props, Params> = async ({ params })
...
-			page: null,
-			publication,
-		},
+		notFound: true,
 		revalidate: 1,
 	};
 };
 
 export async function getStaticPaths() {
-	const data = await request<SlugPostsByPublicationQuery, SlugPostsByPublicationQueryVariables>(
```
Comment (Harsh062) [inline packages/blog-starter-kit/themes/personal/pages/[slug].tsx:250]: @iamandrewluca Encountering an error locally while executing pnpm dev for changes in this branch across all 3 themes (hashnode, personal & enterprise).
```
[next] TypeError: Cannot read properties of undefined (reading 'document')
[next]     at parseRequestExtendedArgs (/Users/harsh/Desktop/hashnode-os/forked_starter-kit/starter-kit/node_modules/.pnpm/graphql-request@6.1.0_graphql@16.8.1/node_modules/graphql-request/build/cjs/parseArgs.js:38:25)
[next]     at request (/Users/harsh/Desktop/hashnode-os/forked_starter-kit/starter-kit/node_modules/.pnpm/graphql-request@6.1.0_graphql@16.8.1/node_modules/graphql-request/build/cjs/index.js:330:72)
```
The functionality is operating without issues in the main branch. Could we please investigate this matter?

File: packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx [hunk 2]
```diff
@@ -192,63 +187,45 @@ export const getStaticProps: GetStaticProps<Props, Params> = async ({ params })
...
-			page: null,
-			publication,
-		},
+		notFound: true,
 		revalidate: 1,
 	};
 };
 
 export const getStaticPaths: GetStaticPaths = async () => {
-	const data = await request<SlugPostsByPublicationQuery, SlugPostsByPublicationQueryVariables>(
```
Comment (Harsh062) [inline packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx:251]: Could you please provide insight into the rationale behind the removal of the Typescript generic checks in this instance?

File: packages/blog-starter-kit/themes/personal/pages/[slug].tsx (hunk 1 above)
Comment (iamandrewluca) [reply packages/blog-starter-kit/themes/personal/pages/[slug].tsx:250]: Hey, @Harsh062! Please check [README Running Locally](https://github.com/Hashnode/starter-kit#running-locally)

File: packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx (hunk 2 above)
Comment (iamandrewluca) [reply packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx:251]: `Query` and `Variables` types are completely inferred from `SlugPostsByPublicationDocument`, they don't need to be explicitly used in generics.

image

File: packages/blog-starter-kit/themes/personal/pages/[slug].tsx (hunk 1 above)
Comment (Harsh062) [reply packages/blog-starter-kit/themes/personal/pages/[slug].tsx:250]: Thanks @iamandrewluca . I did not have the right ENV properties configured. All 3 themes are working fine after adding the appropriate properties in the env variables file.

File: packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx (hunk 2 above)
Comment (Harsh062) [reply packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx:251]: Got it. Thanks for the explanation @iamandrewluca .

**Review by Harsh062 — APPROVED** [review]

**Pull request merged**